### PR TITLE
6174/learning object lifecycle status admin dashboard updates

### PR DESCRIPTION
**Pull request by EVogel1999**
This PR updates the client to reflect the changes in the status updates.  Note that there is a UI bug for rejected status where it doesn't show the red.  If the object is released the editor/admin is automatically sent to the relevancy builder.  If it is rejected then they are sent to the admin dashboard.  Completes story [6174/learning-object-lifecycle-status-admin-dashboard-updates](https://app.shortcut.com/clarkcan/story/6174/learning-object-lifecycle-status-admin-dashboard-updates).

Status icon changes:
Screen Shot 2022-01-20 at 2 24 38 PM

Valid status moves:
Screen Shot 2022-01-20 at 2 25 13 PM

Requesting change reasons (only on accepted major, minor, and rejected moves):
Screen Shot 2022-01-20 at 2 25 32 PM

Another valid status moves:
Screen Shot 2022-01-20 at 2 25 46 PM

Author editing an object that is in accepted major/minor:
Screen Shot 2022-01-20 at 2 28 27 PM

Resubmitting accepted major/minor for review:
Screen Shot 2022-01-20 at 2 28 54 PM

### CODE REVIEW

**Review by pzalep1 — COMMENTED** [review]
Code looks good just a few small wording requests: 
1. Could the Move to review on the author component read submit for review to match more closely with submission verbiage?
2. Can I see a screenshot of the proofing --> released/rejected UI?

**Review by JosueMolinaMorales — APPROVED** [review]

File: src/app/onion/core/learning-object.service.ts [hunk 1]
```diff
@@ -195,12 +194,13 @@ export class LearningObjectService {
     id: string,
     authorUsername: string,
     learningObject: { [key: string]: any },
+    reason?: string,
```
Comment (mfranz1) [inline src/app/onion/core/learning-object.service.ts:197]: Can you just enlighten me why reason is an optional parameter on these routes.

File: src/app/onion/core/learning-object.service.ts [hunk 2]
```diff
@@ -494,6 +494,25 @@ export class LearningObjectService {
...
+  }: {
+    username: string,
+    objectId: string,
+    status: LearningObject.Status,
+    reason?: string
+  }): Promise<void> {
+    await this.http.post<void>(
+      ADMIN_ROUTES.CHANGE_STATUS(username, objectId),
+      { status, reason },
+      // @ts-ignore
```
Comment (mfranz1) [inline src/app/onion/core/learning-object.service.ts:511]: Just curious what error does this suppress?

**Review by mfranz1 — APPROVED** [review]
Looks good I just had two questions

**Comment by EVogel1999** [comment]
Status changes for moving proofing to released/rejected:
Screen Shot 2022-01-21 at 8 42 15 AM

File: src/app/onion/core/learning-object.service.ts (hunk 1 above)
Comment (EVogel1999) [reply src/app/onion/core/learning-object.service.ts:197]: Reason is only needed when you are moving to the accepted major/minor and rejected statuses.  All other statuses don't need them.

File: src/app/onion/core/learning-object.service.ts (hunk 2 above)
Comment (EVogel1999) [reply src/app/onion/core/learning-object.service.ts:511]: For some reason "response-type" isn't a valid optional parameter anymore.  Without it, if we get a 200 the UI would treat it as an error.

**Review by pzalep1 — APPROVED** [review]

**Review by Cwagne17 — APPROVED** [review]

**Pull request merged**